src/Module/BaseApi.php

```diff
@@ -354,44 +355,152 @@ protected static function getUser($contact_id = null)
 	 */
 	protected static function format(string $root_element, array $data)
 	{
-		$return = api_format_data($root_element, self::$format, $data);
+		$return = self::formatData($root_element, self::$format, $data);
 
 		switch (self::$format) {
-			case "xml":
-				header("Content-Type: text/xml");
+			case 'xml':
+				header('Content-Type: text/xml');
 				break;
-			case "json":
-				header("Content-Type: application/json");
+			case 'json':
+				header('Content-Type: application/json');
 				if (!empty($return)) {
 					$json = json_encode(end($return));
 					if (!empty($_GET['callback'])) {
-						$json = $_GET['callback'] . "(" . $json . ")";
+						$json = $_GET['callback'] . '(' . $json . ')';
 					}
 					$return = $json;
 				}
 				break;
-			case "rss":
-				header("Content-Type: application/rss+xml");
+			case 'rss':
+				header('Content-Type: application/rss+xml');
 				$return  = '<?xml version="1.0" encoding="UTF-8"?>' . "\n" . $return;
 				break;
-			case "atom":
-				header("Content-Type: application/atom+xml");
+			case 'atom':
+				header('Content-Type: application/atom+xml');
 				$return = '<?xml version="1.0" encoding="UTF-8"?>' . "\n" . $return;
 				break;
 		}
 
 		return $return;
 	}
 
+	/**
+	 * walks recursively through an array with the possibility to change value and key
+	 *
+	 * @param array    $array    The array to walk through
+	 * @param callable $callback The callback function
+	 *
+	 * @return array the transformed array
+	 */
+	static public function walkRecursive(array &$array, callable $callback)
```

Comment:
This method should be moved to the Array Util class, can't remember its fully qualified name oof the top of my head.